canvas_sdk/data/task.py

```diff
@@ -0,0 +1,51 @@
+from datetime import datetime
+from enum import Enum
+
+from canvas_sdk.data import Model
+from canvas_sdk.data.patient import Patient
+from canvas_sdk.data.staff import Staff
+from canvas_sdk.effects import Effect, EffectType
+
+
+class Task(Model):
+    class Status(Enum):
+        COMPLETED = "Completed"
+        CLOSED = "Closed"
+        OPEN = "Open"
+
+    id: str | None = None
+    assignee: Staff | None = None
+    patient: Patient | None = None
+    title: str | None = None
+    due: datetime | None = None
+    status: Status | None = None
+    comments: "list[TaskComment] | None" = None
+    labels: list[str] | None = None
+
+    def create(self) -> Effect:
```

Comment:
I really like these methods -- making it easier to create effects the right way will be a huge improvement in usability.

The only issue I have with them is the naming and the overall flow -- we should think about that a bit more. When you call "create", you're not really creating a task -- you're making an effect that will create a task.

Also, is it valid to call more than one of these on the same Task? Like call `update` and then `add_comment`?

I'd be open to brainstorm on the design of this if you want.

Reply:
I've been thinking about this more. As a user who is playing around and hasn't read the documentation, I might call `Task.create()` or `Task.update()` and think that is actually mutating the Task.

Some alternatives that could provide a clearer interface:

```
Task.create_effect(...) # clear, but sounds like we're creating an effect rather than creating a "create effect"
Task.make_create_effect(...) # much clearer, but wordy
Task.effect(CREATE, ...) # meh
Task.effect_create(...) # not really that different than the first one
```

Or perhaps Effects should have a class hierarchy, and the payloads should live there:

```
[
    TaskCreateEffect(...),
    TaskUpdateEffect(...),
    TaskAddCommentEffect(...),
]
```

The downside to this would be more classes for a user to learn. This could be possibly addressed by using a class to namespace factory methods around task effects:

```
TaskEffect.create(...)
TaskEffect.update(...)
TaskEffect.add_comment(...)
```

I kind of like that last one. We can still use the shorter names, but it's much clearer that calling "create" doesn't actually create a Task, and the payloads would still be abstracted away from the user.

Reply:
From a code readability and to actually describe what's happening, I see your point with the naming. However, I also see the argument that because this is being built for outside developers, that simply `create`, `update`, etc. make it a more straightforward experience. This is the only control that they will have, and as far as they are concerned, those methods will create and update the models instances. Does it matter that it happens downstream? I talked with AD a bit about this today, and his view are that naming the methods `create`, `update`, etc. are just fine.